test/Libraries/DynamoPythonTests/PythonEvalTests.cs

```diff
@@ -171,6 +172,39 @@ public void CPythonEngineIncludesTraceBack()
             }
         }
 
+        [Test]
+        public void CPythonEngineWithErrorRaisesCorrectEvent()
+        {
+            var code = @"
+# extra line
+1/0
+# extra line
+"; 
+            DSCPython.CPythonEvaluator.EvaluationEnd += CPythonEvaluator_EvaluationEnd;
+            var count = 0;
+            try
+            {
+                DSCPython.CPythonEvaluator.EvaluatePythonScript(code, new ArrayList(), new ArrayList());
+                Assert.Fail("An exception was expected");
+            }
+            catch (Exception exc)
+            {
+                // Trace back is expected here. Line is 3 due to the line break from the code variable declaration
+                Assert.AreEqual(@"ZeroDivisionError : division by zero ['  File ""<string>"", line 3, in <module>\n']", exc.Message);
+                count = count + 1;
+            }
+            finally
+            {
+                DSCPython.CPythonEvaluator.EvaluationEnd -= CPythonEvaluator_EvaluationEnd;
+                Assert.AreEqual(1, count);
+            }
+        }
```

Comment:
This is not a good test because:
1. Even if the event was not fired it would still pass.
2. Does not test the event gives a success state when it runs the code without errors.

On a secondary level, this looks like copied from another test that checked the line number of the traceback. That doesn't help to see the intention of the test clearly, so I would recommend removing the non-relevant parts.

Reply:
@mmisol thanks, I put the count assertion in the wrong place, will fix it up.

Reply:
actually @mmisol - your other points are still valid, but I think 1. is false.
If this event was not raised, then the assertion of  count == 1 in the finally block would fail the test, would it not?

Reply:
`count` is increased in the `catch` block, not by the handler, right? My point is, how would you know the handler was executed if it does not change anything outside of himself?

Reply:
👍